pkg/networkfs/resolvconf/resolvconf.go

```diff
@@ -65,17 +77,30 @@ func GetLastModified() ([]byte, string) {
 	return lastModified.contents, lastModified.sha256
 }
 
-// RemoveReplaceLocalDns looks for localhost (127.*) entries in the provided
-// resolv.conf, removing local nameserver entries, and, if the resulting
-// cleaned config has no defined nameservers left, adds default DNS entries
+// FilterResolvDns has two main jobs:
+// 1. It looks for localhost (127.*|::1) entries in the provided
+//    resolv.conf, removing local nameserver entries, and, if the resulting
+//    cleaned config has no defined nameservers left, adds default DNS entries
+// 2. Given the caller provides the enable/disable state of IPv6, the filter
+//    code will remove all IPv6 nameservers if it is not enabled for containers
+//
 // It also returns a boolean to notify the caller if changes were made at all
-func RemoveReplaceLocalDns(resolvConf []byte) ([]byte, bool) {
+func FilterResolvDns(resolvConf []byte, ipv6Enabled bool) ([]byte, bool) {
 	changed := false
-	cleanedResolvConf := localHostRegexp.ReplaceAll(resolvConf, []byte{})
-	// if the resulting resolvConf is empty, use defaultDns
-	if !bytes.Contains(cleanedResolvConf, []byte("nameserver")) {
-		log.Infof("No non-localhost DNS nameservers are left in resolv.conf. Using default external servers : %v", defaultDns)
-		cleanedResolvConf = append(cleanedResolvConf, []byte("\nnameserver "+strings.Join(defaultDns, "\nnameserver "))...)
+	cleanedResolvConf := localhostRegexp.ReplaceAll(resolvConf, []byte{})
+	// if IPv6 is not enabled, also clean out any IPv6 address nameserver
+	if !ipv6Enabled {
+		cleanedResolvConf = nsIPv6Regexp.ReplaceAll(cleanedResolvConf, []byte{})
+	}
+	// if the resulting resolvConf has no more nameservers defined, add appropriate
+	// default DNS servers for IPv4 and (optionally) IPv6
+	if len(GetNameservers(cleanedResolvConf)) == 0 {
+		log.Infof("No non-localhost DNS nameservers are left in resolv.conf. Using default external servers : %v", defaultIPv4Dns)
+		cleanedResolvConf = append(cleanedResolvConf, []byte("\n"+strings.Join(defaultIPv4Dns, "\n"))...)
```

Comment:
I think it's cleaner to combine array of needed dns, like `dns := append(defaultIPv4Dns, defaultIPv6Dns)` and then use all your `Join` magic.